src/api/messageprocessors/ListingItemMessageProcessor.ts

```diff
@@ -7,22 +8,63 @@ import { ListingItemFactory } from '../factories/ListingItemFactory';
 import { ListingItemService } from '../services/ListingItemService';
 import { ListingItem } from '../models/ListingItem';
 
+import { ItemCategoryFactory } from '../factories/ItemCategoryFactory';
+import { MessagingInformationFactory } from '../factories/MessagingInformationFactory';
+import { ItemPriceFactory } from '../factories/ItemPriceFactory';
+import { ItemCategoryService } from '../services/ItemCategoryService';
+import { isArray } from 'util';
+
 export class ListingItemMessageProcessor implements MessageProcessorInterface {
 
     public log: LoggerType;
     constructor(
         @inject(Types.Factory) @named(Targets.Factory.ListingItemFactory) public listingItemFactory: ListingItemFactory,
+        @inject(Types.Factory) @named(Targets.Factory.ItemCategoryFactory) public itemCategoryFactory: ItemCategoryFactory,
+        @inject(Types.Factory) @named(Targets.Factory.MessagingInformationFactory) public mesInfoFactory: MessagingInformationFactory,
+        @inject(Types.Factory) @named(Targets.Factory.ItemPriceFactory) public itemPriceFactory: ItemPriceFactory,
         @inject(Types.Service) @named(Targets.Service.ListingItemService) public listingItemService: ListingItemService,
+        @inject(Types.Service) @named(Targets.Service.ItemCategoryService) public itemCategoryService: ItemCategoryService,
         @inject(Types.Core) @named(Core.Logger) public Logger: typeof LoggerType
     ) {
         this.log = new Logger(__filename);
     }
 
     @validate()
     public async process(message: any): Promise<ListingItem> {
+        // get Category
+        const rootCategoryWithRelated: any = await this.itemCategoryService.findRoot();
+        const itemCategory: any = await this.itemCategoryFactory.get(message.information.category, rootCategoryWithRelated);
+        let itemCategoryId;
+        // create category if need
+        if (isArray(itemCategory)) {
```

Comment:
this is quite long function so this could be a separate function to make this readable again.
createUserDefinedItemCategoriesOrSomething(itemCategory)